common/network-common/src/main/java/org/apache/spark/network/server/TransportRequestHandler.java

```diff
@@ -127,6 +127,7 @@ private void processStreamRequest(final StreamRequest req) {
     ManagedBuffer buf;
     try {
       buf = streamManager.openStream(req.streamId);
+      streamManager.registerChannel(channel, req.streamId);
```

Comment:
I'm wondering when we should do this. There are many kinds of requests, and currently only chunk fetch request does it.

Reply:
For stream request and get chunk request, they are recorded in stream manager. Registering channel for them is to make sure they are removed from stream manager when the channel is inactive. For other types of requests, I don't find they are recorded like that.